bom/pom.xml

```diff
@@ -740,9 +742,21 @@
       <dependency>
         <groupId>com.squareup.okhttp3</groupId>
         <artifactId>okhttp</artifactId>
-        <version>4.9.0</version>
+        <version>${okhttp.version}</version>
         <scope>runtime</scope>
       </dependency>
+      <dependency>
+        <groupId>com.squareup.okhttp3</groupId>
+        <artifactId>logging-interceptor</artifactId>
```

Comment:
is this really a *required* dependency? Or can we also omit this?

Reply:
The OkHttp `logging-interceptor` artifact in version 3.14.9 is used by the `kubernetes-client` dependency.
Excluding the entry here would mean that we use `logging-interceptor:3.14.9` along with `okhttp:4.9.0`.
I would rather use one version for the OkHttp components.

I've therefore amended the commit to make this more explicit, excluding the OkHttp dependencies from `kubernetes-client` and `kubernetes-server-mock` dependency declarations and explicitly adding the OkHttp dependencies in the POM where the Kubernetes client/server-mock dependencies get used.

Reply:
> Excluding the entry here would mean that we use logging-interceptor:3.14.9 along with okhttp:4.9.0.
I would rather use one version for the OkHttp components.

???

My question was if the kubernetes-client would also work if this artifact were not on the class path.

Reply:
No, it wouldn't work.
Exception in the Command Router component:
```
java.lang.ClassNotFoundException: okhttp3.logging.HttpLoggingInterceptor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:104)
	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:60)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:145)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:66)
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:51)
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:43)
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.<init>(DefaultKubernetesClient.java:138)
	at org.eclipse.hono.commandrouter.impl.KubernetesBasedAdapterInstanceStatusService.<init>(KubernetesBasedAdapterInstanceStatusService.java:72)
```
`kubernetes-server-mock` used in the unit test also needs it:
```
java.lang.NoClassDefFoundError: okhttp3/logging/HttpLoggingInterceptor
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:145)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClientForMockServer(HttpClientUtils.java:70)
	at io.fabric8.kubernetes.client.server.mock.KubernetesMockServer.createClient(KubernetesMockServer.java:91)
	at io.fabric8.kubernetes.client.server.mock.KubernetesMockServerExtension.initializeKubernetesClientAndMockServer(KubernetesMockServerExtension.java:92)
	at io.fabric8.kubernetes.client.server.mock.KubernetesMockServerExtension.setKubernetesClientAndMockServerFields(KubernetesMockServerExtension.java:123)
	at io.fabric8.kubernetes.client.server.mock.KubernetesMockServerExtension.beforeAll(KubernetesMockServerExtension.java:72)
...
```

Reply:
that's a pity